controller/metrics/clustercollector.go

```diff
@@ -54,9 +56,11 @@ type HasClustersInfo interface {
 }
 
 type clusterCollector struct {
-	infoSource HasClustersInfo
-	info       []cache.ClusterInfo
-	lock       sync.Mutex
+	infoSource    HasClustersInfo
+	info          []cache.ClusterInfo
+	lock          sync.Mutex
+	clusters      argoappv1.ClusterList
```

Comment:
I will need to review more in details, but adding clusters here is to get the initial labels, however, the clusters are never updated and data will be stale if the labels change.

I think there are 2 possible implementations
- the `HasClustersInfo` needs to be updated to add a `GetLabels` function. Then the Collect can use that on the (based on the non-stale stateCache) to get the labels. Note: I dont know if this information is available in the stateCache
- Similar to how it is done for Applications, pass the cluster lister to the clusterCollector so it can fetch the current data when collect is called.

Reply:
@agaudreault 
thanks
okay, which one do you think is the best way to do it ?

Reply:
Option 2 seems to follow the actual pattern. Not sure if option 1 is feasible.